src/main/java/org/jnode/fs/xfs/inode/INode.java

```diff
@@ -393,27 +397,70 @@ public String getSymLinkText() {
      *
      * @return list of inode attributes.
      */
-    public List<XfsAttribute> getAttributes() throws IOException {
+    public List<FSAttribute> getAttributes() throws IOException {
         long off =  getOffset() + getINodeSizeForOffset() + (getAttributesForkOffset() * 8);
         final XfsAttributeHeader myXfsAttributeHeader = new XfsAttributeHeader(getData(), off);
         final long attributesFormat = getAttributesFormat();
         if (attributesFormat == 1) {
             off += 4;  // header length remeber header has a 1 byte padding
             final int count = (int) myXfsAttributeHeader.getCount();
-            List<XfsAttribute> attributes = new ArrayList<>(count);
+            List<FSAttribute> attributes = new ArrayList<>(count);
             for (int i = 0; i < count; i++) {
                 final XfsAttribute attribute = new XfsAttribute(getData(), off);
                 attributes.add(attribute);
                 off += attribute.getAttributeSizeForOffset();
             }
             return attributes;
+        } else if (attributesFormat == 2) {
```

Comment:
I'm guessing that `attributesFormat` corresponds to `typedef enum xfs_dinode_fmt`. I guess this corresponds to the `INodeFormat` enum above? Can you use that here?

Reply:
Deleted the enum `INodeFormat` and used the constants in `XfsConstants` for consistency